topo_processor/stac/item_factory.py

```diff
@@ -6,34 +6,30 @@
 from topo_processor.data.data_transformers import data_transformer_repo
 from topo_processor.metadata.metadata_loaders import metadata_loader_repo
 from topo_processor.metadata.metadata_validators import metadata_validator_repo
-from topo_processor.stac import DataType, Item
-from topo_processor.stac.collection_store import get_collection
+from topo_processor.stac.store import get_asset, item_store
 from topo_processor.util import time_in_ms
 
 
-async def create_items(source_dir: str, data_type: DataType, target: str, temp_dir: str) -> None:
+async def create_items(source_dir: str) -> None:
     start_time = time_in_ms()
-    items_to_process = []
+
+    assets_to_process = []
     for file_ in os.listdir(source_dir):
-        source_path_path = os.path.join(source_dir, file_)
-        item = Item(source_path_path, data_type, target, temp_dir)
+        path = os.path.join(source_dir, file_)
+        asset = get_asset(path)
+        assets_to_process.append(metadata_loader_repo.load_metadata(asset))
+    await asyncio.gather(*assets_to_process)
+
+    items_to_process = []
+    for item in item_store.values():
         items_to_process.append(process_item(item))
     await asyncio.gather(*items_to_process)
+
     get_log().debug("Items Created", source_dir=source_dir, duration=time_in_ms() - start_time)
 
 
 async def process_item(item):
-    await metadata_loader_repo.load_metadata(item)
-    if item.is_valid:
+    if item.check_validity:
         await metadata_validator_repo.validate_metadata(item)
-    if item.is_valid:
+    if item.check_validity:
```

Comment:
I reckon change this back to is_valid.. more comment on item.py

Reply:
I also prefer is_valid, however then the item with have both a function and a variable with the name name which causes errors:
`TypeError: 'bool' object is not callable`